bin/testfiles/TEST_HTTP_HttpClient4.xml

```diff
@@ -144,7 +144,7 @@
   <httpSample s="true" lb="HR-FollowRedirect-18" rc="302" dt="text"/>
   <httpSample s="true" lb="HR-FollowRedirect-19" rc="302" dt="text"/>
   <httpSample s="true" lb="HR-FollowRedirect-20" rc="302" dt="text"/>
-  <httpSample s="false" lb="HR-FollowRedirect-21" rc="Non HTTP response code: java.io.IOException" dt="text"/>
```

Comment:
I don't understand this modification. Why 20 instead of 21 ?

Reply:
If you expect `HR-FollowRedirect-21` here we can  fix it using this small hack:
```
if (redirect >= MAX_REDIRECTS) {
            lastRes = errorResult(new IOException("Exceeded maximum number of redirects: " + MAX_REDIRECTS), new HTTPSampleResult(lastRes));
            lastRes.setSampleLabel(totalRes.getSampleLabel()); // use parent label 
            totalRes.addSubResult(lastRes);
}
```